### PR TITLE
Refactor validation/admission webhook

### LINKED ISSUE #38: Refactor the validation webhook to the GCP provider extension model
The current validation webhook (or "validator") introduced with #11 is based on the model that could currently be found in most other provider extensions, e.g. the [vSphere provider extension](https://github.com/gardener/gardener-extension-provider-vsphere). However, a new model has been introduced and some refactoring is in progress in this area currently. The [GCP provider extension](https://github.com/gardener/gardener-extension-provider-gcp) has already been refactored to this new model. Using it as an example, we should refactor the Kubevirt validation webhook to this model as well.

/area usability
/kind technical-debt
/priority normal
/platform kubevirt

### PR DESCRIPTION
**Pull request by moadqassem**
**How to categorize this PR?**
/area usability
/kind technical-debt
/priority normal
/platform kubevirt

**What this PR does / why we need it**:
Refactors the validation webhook in order to follow the GCP provider extension model. 

**Which issue(s) this PR fixes**:
Fixes #38

**Special notes for your reviewer**:

**Release note**:
```improvement operator
NONE
```

### CODE REVIEW

**Review by stoyanr — APPROVED** [review]

File: charts/gardener-extension-admission-kubevirt/charts/application/templates/validatingwebhook-validator.yaml [hunk 1]
```diff
@@ -29,3 +30,27 @@ webhooks:
       path: /webhooks/validate
     {{- end }}
     caBundle: {{ required ".Values.webhookConfig.caBundle is required" (b64enc .Values.global.webhookConfig.caBundle) }}
+- name: secrets.validation.gcp.provider.extensions.gardener.cloud
```
Comment (mfranczy) [inline charts/gardener-extension-admission-kubevirt/charts/application/templates/validatingwebhook-validator.yaml:33]: I think that should be `secrets.validation.kubevirt.provider.extensions.gardener.cloud`

**Review by mfranczy — CHANGES_REQUESTED** [review]
The PR looks good to me, I found one minor thing.

File: charts/gardener-extension-admission-kubevirt/charts/application/templates/validatingwebhook-validator.yaml (hunk 1 above)
Comment (stoyanr) [reply charts/gardener-extension-admission-kubevirt/charts/application/templates/validatingwebhook-validator.yaml]: Thanks for noticing, fixed.

**Review by mfranczy — APPROVED** [review]
/lgtm

**Pull request merged**